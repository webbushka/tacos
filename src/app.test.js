import React from 'react';
import { shallow } from 'enzyme';
import App from './app';

const wrapper = shallow(<App />);

describe('Component: app.js', () => {
  it('should render without errors', () => {
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper).toHaveTagName('div');
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the correct initial state', () => {
    const state = {
      tacos: [
        { name: 'Asada', src: 'asada.png' },
        { name: 'Carnitas', src: 'carnitas.png' },
        { name: 'Pollo', src: 'pollo.png' },
      ],
    };
    expect(wrapper.state()).toEqual(state);
  });

  it('should should have the correct children', () => {
    const tacos = wrapper.state('tacos');
    const toggles = wrapper.find('ContentToggle');
    expect(toggles.length).toBe(3);
    toggles.forEach((toggle, ind) => {
      expect(toggle).toHaveProp('summary', tacos[ind].name);
      expect(toggle.key()).toBe(tacos[ind].name);
      expect(toggle.childAt(0)).toHaveTagName('img');
    });
  });

  it('should have correct props on child image', () => {
    const tacos = wrapper.state('tacos');
    const toggles = wrapper.find('ContentToggle');
    toggles.forEach((toggle, ind) => {
      const img = toggle.find('img');
      expect(img).toHaveProp('src', tacos[ind].src);
      expect(img).toHaveProp('alt', `${tacos[ind].name} taco`);
    });
  });
});
