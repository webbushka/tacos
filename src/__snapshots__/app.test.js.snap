// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: app.js should match the snapshot 1`] = `
<div>
  <ContentToggle
    key="Asada"
    summary="Asada"
  >
    <img
      alt="Asada taco"
      className="taco"
      src="asada.png"
    />
  </ContentToggle>
  <ContentToggle
    key="Carnitas"
    summary="Carnitas"
  >
    <img
      alt="Carnitas taco"
      className="taco"
      src="carnitas.png"
    />
  </ContentToggle>
  <ContentToggle
    key="Pollo"
    summary="Pollo"
  >
    <img
      alt="Pollo taco"
      className="taco"
      src="pollo.png"
    />
  </ContentToggle>
</div>
`;
